File: VB/ViewModels/ModuleAViewModel.vb
Imports Microsoft.VisualBasic
Imports System
Namespace WindowsFormsApplication1.ViewModels
    Public Class ViewModelModuleA
        Public Property AllowAnotherAction() As Boolean
            Get
                Return m_AllowAnotherAction
            End Get
            Private Set(value As Boolean)
                m_AllowAnotherAction = Value
            End Set
        End Property
        Private m_AllowAnotherAction As Boolean
        Friend Sub DoSomeAction()
            AllowAnotherAction = True
        End Sub
        Friend Sub ResetSomeAction()
            AllowAnotherAction = False
        End Sub
    End Class
End Namespace

File: CS/WindowsFormsApplication1/ViewModels/ModuleAViewModel.cs
namespace WindowsFormsApplication1.ViewModels {
    public class ViewModelModuleA {
        public bool AllowAnotherAction { get; private set; }
        internal void DoSomeAction() {
            AllowAnotherAction = true;
        }
        internal void ResetSomeAction() {
            AllowAnotherAction = false;
        }
    }
}